src/screens/Profiles/components/MediaDetails/component.js

```diff
@@ -0,0 +1,200 @@
+const TITLE_MAP = {
+  en: 'English',
+  en_us: 'English',
+  en_jp: 'Romanized',
+  ja_jp: 'Japanese',
+  en_cn: 'Romanized',
+  zh_cn: 'Chinese',
+  en_th: 'Romanized',
+  th_th: 'Thai',
+  en_kr: 'Romanized',
+  ko_kr: 'Korean',
+};
+
+export class MediaDetails extends Component {
+  static propTypes = {
+    media: PropTypes.object.isRequired,
+  };
+
+  static defaultProps = {
+    media: null,
+  };
+
+  state = {
+    isViewingMore: false,
+  };
+
+  constructor(props) {
+    super(props);
+    this.isAnime = props.media.type === 'anime';
+  }
+
+  /**
+   * Don't re-render when `media` updates as we have all relevant information
+   * on the initial request.
+   */
+  shouldComponentUpdate(nextProps, nextState) {
+    return nextState.isViewingMore !== this.state.isViewingMore;
+  }
+
+  getStudios = () => {
+    const { media: { animeProductions } } = this.props;
+    let productions = animeProductions || [];
+    productions = productions.filter(item => item.role === 'studio');
+    return productions.length > 0 ?
+      productions
+        .map(item => item.producer && item.producer.name)
+        .join(', ')
+      : 'TBD';
+  };
+
+  getUnitCount = () => (
+    this.isAnime ? this.props.media.episodeCount : this.props.media.chapterCount
+  );
+
+  getUnitLength = () => {
+    if (!this.isAnime) { return null; }
+    const count = this.getUnitCount();
+    if (!count || isNull(this.props.media.episodeLength)) { return null; }
+    return count > 1 ? ` (${this.props.media.episodeLength} minutes each)` :
+      ` (${this.props.media.episodeLength} minutes)`;
+  };
+
+  getSeason = date => {
+    const month = date.month() + 1;
+    let season = '';
+    if ([12, 1, 2].includes(month)) {
+      season = 'Winter';
+    } else if ([3, 4, 5].includes(month)) {
+      season = 'Spring';
+    } else if ([6, 7, 8].includes(month)) {
+      season = 'Summer';
+    } else if ([9, 10, 11].includes(month)) {
+      season = 'Fall';
+    }
+    let year = date.year();
+    // December is considered Winter of the next year
+    if (season === 'Winter' && month === 12) {
+      year = year + 1;
+    }
+    return `${season} ${year}`;
+  };
+
+  getStartDate = () => {
+    const date = moment(this.props.media.startDate);
+    const season = this.getSeason(date);
+    const formattedDate = date.format('MMM D');
+    return `${formattedDate}, ${season}`;
+  };
+
+  getAgeRating = () => (
+    isNull(this.props.media.ageRatingGuide) ?
+      this.props.media.ageRating
+      : `${this.props.media.ageRating} - ${this.props.media.ageRatingGuide}`
+  );
+
+  getMediaTitles = () => {
+    const { media: { titles } } = this.props;
+    const map = [];
+    Object.keys(titles).forEach(key => {
+      map.push({ key: TITLE_MAP[key], value: titles[key] });
+    });
+    return map;
+  };
+
+  onViewMore = () => {
+    const isViewingMore = !this.state.isViewingMore;
+    this.setState({ isViewingMore });
+  };
+
+  render() {
+    const { media } = this.props;
```

Comment:
should validate it to only render the info if `media` is not null.
Putting the validation here will mean the other functions above shouldn't throw an error when `media` is null.